File: 2D.c
#include <stdio.h>
#include <time.h>

void main(){
  system("clear");
  printf("\033[99m"); //Set color.
  int H = 30;
  int W = 30;
  int Count = 0;
  int AX, AY, POINT;
  int Gird[H][W];
  int Temp[H][W];
  int i, j, k, l, a, b; //Set values for "for".
  
  srand(time(NULL));
  for (a = 0; a < H; a++) {
    for (b = 0; b < W; b++) {
      Gird[a][b] = rand() % 2; //Generate random point.
    }
  }
  /*
  for(int k = 3; k < W; k=k+6){  //Generate many flight.
    for(int l = 2; l < H; l=l+5){
      Gird[l][k] = 1;
      Gird[l][k+1] = 1;
      Gird[l][k+2] = 1;
      Gird[l+1][k] = 1;
      Gird[l+2][k+1] = 1;
    }
  }
  */
  while (1) {
    printf("\x1b[0;0H");
    //Generate new map.
    for (a = 0; a < H; a++) {
      for (b = 0; b < W; b++) {
	POINT = Gird[a][b];
	//Get neighbour.
	Count = - POINT;
	for (i = -1; i < 2; i++) {
	  AY = (a + i + H) % H;
	  for (j = -1; j < 2; j ++) {
	    AX = (b + j + W) % W;
	    Count += Gird[AY][AX];
	  }
	}
	//Get new point.
	if (POINT && (Count < 2 || Count > 3)) {
	  Temp[a][b] = 0;
	} else if (!POINT && Count == 3){
	  Temp[a][b] = 1;
	} else {
	  Temp[a][b] = POINT;
	}
	//Draw to screen.
	if (Gird[a][b]) {
	  printf("# ");
        } else {
	  printf("  ");
	}
      }
      printf("\n");
    }
    //Flush the map and empty the temp.
    for (a = 0; a < H; a++) {
      for (b = 0; b < W; b++) {
	Gird[a][b] = Temp[a][b];
	Temp[a][b] = 0;
      }
    }
  }
}

File: 2d.py
import time
import random
import os
from copy import deepcopy as dp

def GetGird(H, W):
    Gy = []
    Gird = []
    for i in range(W): Gy += [0]
    for i in range(H): Gird += [dp(Gy)]
    return Gird

def GetRandom(H, W, Gird):
    for y in range(H):
        for x in range(W):
            Gird[y][x] = random.randint(0,1)
    return Gird

def GetCount(Gird, x, y, H, W):
    Point = Gird[y][x]
    Count = -Point
    for i in range(-1, 2):
        for j in range(-1, 2):
            AY = (y + i + H) % H
            AX = (x + j + W) % W
            Count += Gird[AY][AX]
    return Count

def Generate(Lifes, Templ, H, W):
    for y in range(H):
        for x in range(W):
            Count = GetCount(Lifes, x, y, H, W)
            if Lifes[y][x] and ((Count < 2) or (Count > 3)):
                Templ[y][x] = 0
            elif not(Lifes[y][x]) and (Count == 3):
                Templ[y][x] = 1
            else:
                Templ[y][x] = Lifes[y][x]
    return dp(Templ)

def Draw(Gird, H, W):
    print("\x1b[0;0H")
    for y in range(H):
        Screen = ""
        for x in range(W):
            if Gird[y][x]: Screen += "# "
            else: Screen += "  "
        print(Screen)

os.system("clear") # On Linux
#os.system("cls") # On Windows
print("\033[1;99m Game of Life!")
W = 30
H = 30
Templ = GetGird(H, W)
Lifes = GetGird(H, W)
Lifes = GetRandom(H, W, Lifes)
while 1:
    Draw(Lifes, H, W)
    Lifes = Generate(Lifes, Templ, H, W)
 
